src/components/Popup.js

```diff
@@ -37,5 +39,12 @@ export default class Popup {
         this.close();
       }
     });
+    this._popupElement.addEventListener("keydown", (event) => {
+      this._handleEscKeyPress(event);
+    });
+
+    this._popupElement.addEventListener("click", (event) => {
+      this._handleModalOverlay(event);
+    });
```

Comment:
Please, delete this new code of overlay and Escape handling because you do everything in `open` and in line 38